src/shared/conf-parser.c

```diff
@@ -221,8 +221,19 @@ static int parse_line(
                         return -ENOMEM;
 
                 if (sections && !nulstr_contains(sections, n)) {
+                        bool ignore = flags & CONFIG_PARSE_RELAXED;
+                        const char *t;
 
-                        if (!(flags & CONFIG_PARSE_RELAXED) && !startswith(n, "X-"))
+                        ignore = ignore || startswith(n, "X-");
+
+                        if (!ignore)
+                                NULSTR_FOREACH(t, sections)
+                                        if (streq_ptr(n, startswith(t, "-"))) {
+                                                ignore = true;
+                                                break;
+                                        }
+
+                        if (!ignore)
                                 log_syntax(unit, LOG_WARNING, filename, line, 0, "Unknown section '%s'. Ignoring.", n);
 
                         free(n);
```

Comment:
hmm, thinking about this again, what is the actual difference between sections listed with and without "-" here now? I mean, we don't complain for anything listed without the "-", and we also don't complain for anything listed with the "-", so what's the point?

Reply:
For unit files, we know the file type before we start parsing, because it is in the extension. For netdev files, we first parse with a partial parser and just the header allocated, check the conditions, and then we parse a second time with the full structure size allocated. So we need to be able to tell the parser which sections should be silently ignored in the first round.

BTW, I tried to explain this in the commit message...